scripts/globals/combat/physical_utilities.lua

```diff
@@ -684,3 +684,298 @@ end
+    local defenderAgi = defender:getStat(xi.mod.AGI)
+
+    guardRate = utils.clamp(((guardSkill * 0.1 + (defenderAgi - attackerDex) * 0.125 + 10) * levelDiffMult), 5, 25)
+
+    return guardRate
+end
+
+xi.combat.physical.canBlock = function(defender, attacker)
+    local canBlock = false
+
+    if defender:isFacing(attacker) and not defender:hasPreventActionEffect() then
+        if defender:isPC() and defender:getSkillRank(xi.skill.SHIELD) > 0 then
+            local shield = defender:getEquippedItem(xi.slot.SUB)
+            if shield then
+                canBlock = shield:isShield()
+            end
+        elseif
+            defender:isMob() or
+            defender:isPet() or
+            defender:isTrust()
+        then
+            canBlock = defender:getMobMod(xi.mobMod.CAN_SHIELD_BLOCK) > 0
+        end
+    end
+
+    return canBlock
+end
+
+xi.combat.physical.calculateBlockRate = function(defender, attacker)
+    local blockRate = 0
+    local shieldSize = 3
+    local skillModifier = 0
+    local palisadeMod = defender:getMod(xi.mod.PALISADE_BLOCK_BONUS)
+    local reprisalMult = 1.0
+
+    -- assume bare hands case
+    local attackerSkillType = xi.skill.HAND_TO_HAND
+    if not attacker:isUsingH2H() then
+        attackerSkillType = attacker:getWeaponSkillType(xi.slot.MAIN)
+    end
+
+    local attackSkill = attacker:getSkillLevel(attackerSkillType)
+    local blockSkill = defender:getSkillLevel(xi.skill.SHIELD)
+
+    if defender:isPC() then
+        local shield = defender:getEquippedItem(xi.slot.SUB)
+        -- already checked in canBlock but check again here to make sure
+        if shield and shield:isShield() then
+            shieldSize = shield:getShieldSize()
+        else
+            return 0
+        end
+    elseif
+        defender:isMob() or
+        defender:isPet() or
+        defender:isTrust()
+    then
+        -- already checked in canBlock but check again here to make sure
+        if defender:getMobMod(xi.mobMod.CAN_SHIELD_BLOCK) > 0 then
+            blockRate = defender:getMod(xi.mod.SHIELDBLOCKRATE)
+            -- automations are a special case
+            if defender:isAutomaton() then
+                skillModifier = (defender:getSkillLevel(xi.skill.AUTOMATON_MELEE) - attackSkill) * 0.215
+                return math.max(0, blockRate + skillModifier)
+            -- mobs and trusts use max skill for job and level
+            elseif defender:isTrust() then
+                -- TODO: check trust type for ilvl > 99 when implemented
+                blockSkill = defender:getMaxSkillLevel(math.min(defender:getMainLvl(), 99), defender:getMainJob(), xi.skill.SHIELD)
+            else
+                blockSkill = defender:getMaxSkillLevel(defender:getMainLvl(), defender:getMainJob(), xi.skill.SHIELD)
+            end
+        else -- No block mobmod so zero rate
+            return 0
+        end
+    end
+
+    if defender:isPC() then
+        if shieldSize == 1 then -- Buckler
```

Comment:
I would probably table the blockrates and do a lookup based on shield size instead of a long condition

Reply:
> I would probably table the blockrates and do a lookup based on shield size instead of a long condition

Thanks for the suggestion, I updated the PR!